Assets/Scripts/DataCollection/PlayerIdentifier.cs

```diff
@@ -3,17 +3,31 @@
 using UnityEngine;
 using System.Runtime.InteropServices;
 using UnityEngine.UI;
+using Photon.Pun.Demo.PunBasics;
 
 
 public class PlayerIdentifier : MonoBehaviour
 {
     [DllImport("__Internal")]
     public static extern void RequestPlayerIdentifier();
 
-    public string playerIdentifier;
+    public string playerIdentifier = "";
 
     void GetPlayerIdentifier(string data)
     {
         playerIdentifier = data;
     }
+
+    public void SetPlayerIdentifiers()
+    {
+        List<GameObject> players = new List<GameObject>(GameObject.FindGameObjectsWithTag("Player"));
+        foreach (GameObject player in players)
+        {
+            if (player.GetComponent<PlayerManager>().IsPlayerLocal())
+            {
+                player.GetComponent<PlayerManager>().CallGetPlayerIdentifier(playerIdentifier);            //we send other players our player identifier
+            }
+
+        }
+    }
 }
```

Comment:
Can't you use `PlayerManager.LocalPlayerManager` here?

Reply:
Not sure, but it would need retesting, and this function is only called once, so dont want to mess around with it. Unless you are sure it would work?

Reply:
Just this seems very redundant code cause if you're finding all the players in game, one of them will definitely be local, so idk. Let's leave it I guess if it's just used once, but for the next time we should remember we have better options easily accessible